### PR TITLE
Use track subnets flag

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-network-runner/issues/414 https://github.com/ava-labs/avalanche-network-runner/issues/415
It is not changing external RPC + Client usage of whitelisted subnets, only avago flags and internal data structures

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -383,9 +383,15 @@ func WithExecPath(execPath string) OpOption {
 	}
 }
 
-func WithWhitelistedSubnets(whitelistedSubnets string) OpOption {
+func WithWhitelistedSubnets(trackSubnets string) OpOption {
```
Comment (holisticode) [inline client/client.go:386]: Why are we keeping this one?

File: local/blockchain.go [hunk 2]
```diff
@@ -423,41 +423,41 @@ func (ln *localNetwork) getCurrentSubnets(ctx context.Context) ([]ids.ID, error)
...
 // TODO: make this "restart" pattern more generic, so it can be used for "Restart" RPC
-func (ln *localNetwork) restartNodesWithWhitelistedSubnets(
+func (ln *localNetwork) restartNodesWithTrackSubnets(
 	ctx context.Context,
 	subnetIDs []ids.ID,
 ) (err error) {
 	fmt.Println()
-	ln.log.Info(logging.Green.Wrap("restarting each node"), zap.String("whitelisted-subnets", config.WhitelistedSubnetsKey))
-	whitelistedSubnetIDsMap := map[string]struct{}{}
+
```
Comment (holisticode) [inline local/blockchain.go:431]: I suggest to keep this log but in Debug

File: local/blockchain.go [hunk 3]
```diff
@@ -423,41 +423,41 @@ func (ln *localNetwork) getCurrentSubnets(ctx context.Context) ([]ids.ID, error)
...
+	ln.flags[config.TrackSubnetsKey] = trackSubnets
 
 	for nodeName, node := range ln.nodes {
 		// delete node specific flag so as to use default one
 		nodeConfig := node.GetConfig()
-		delete(nodeConfig.Flags, config.WhitelistedSubnetsKey)
+		delete(nodeConfig.Flags, config.TrackSubnetsKey)
 
-		ln.log.Info("removing and adding back the node for whitelisted subnets", zap.String("node-name", nodeName))
+		ln.log.Info("removing and adding back the node for track subnets", zap.String("node-name", nodeName))
```
Comment (holisticode) [inline local/blockchain.go:460]: This could maybe also be a Debug?

File: scripts/tests.e2e.sh [hunk 4]
```diff
@@ -3,15 +3,15 @@ set -e
 
 export RUN_E2E="true"
 # e.g.,
-# ./scripts/tests.e2e.sh 1.9.5 1.9.4 0.4.7
+# ./scripts/tests.e2e.sh 1.9.7 1.9.6 0.4.8
```
Comment (holisticode) [inline scripts/tests.e2e.sh:6]: Maybe we don't need to always have to update these, as they are just examples. Alternatively move the `DEFAULT_` vars before this block, then we can also put

`# ./scripts/tests.e2e.sh $DEFAULT_VERSION1 $DEFAULT_VERSION2 $DEFAULT_SUBNET_EVM_VERSION` 

Just a suggestion

**Review by holisticode — COMMENTED** [review]
Basically lgtm, just a couple of Qs

File: client/client.go (hunk 1 above)
Comment (felipemadero) [reply client/client.go:386]: For back compatibility. That is part of the ANR client library API (eg used by CIs). I think we can keep it and plan to remove it in a couple of releases in the future.

File: local/blockchain.go (hunk 2 above)
Comment (felipemadero) [reply local/blockchain.go:431]: this log was kinda useless. it is printing the name of config.WhitelistedSubnetsKey flag

Comment (holisticode) [reply local/blockchain.go:431]: It was also printing "restarting each node" which seems useful in Debug. Nevertheless, leave it up to you.

Comment (felipemadero) [reply local/blockchain.go:431]: We have a very similar log some lines afterwards, which also prints the tracked ids

**Review by holisticode — APPROVED** [review]

**Pull request merged**